src/stores/formula_fingerprints_store.ts

```diff
@@ -0,0 +1,294 @@
+   **/
+  dy: number;
+  /**
+   * Indicates if the reference targets another sheet (0 for same sheet, 1 for another sheet).
+   */
+  crossSheet: number;
+  /**
+   * A literal number (1 if present, 0 otherwise).
+   */
+  number: number;
+  /**
+   * A literal string (1 if present, 0 otherwise).
+   */
+  string: number;
+  /**
+   * A literal boolean (1 if present, 0 otherwise).
+   */
+  boolean: number;
+};
+
+/**
+ * A string that represents a reference vector.
+ */
+type Fingerprint = string;
+
+export class FormulaFingerprintStore extends SpreadsheetStore {
+  mutators = ["enable", "disable"] as const;
+  private isInvalidated = false;
+  isEnabled: boolean = false;
+  map = new PositionMap<Color>();
+
+  handle(cmd: Command) {
+    if (isCoreCommand(cmd) && this.isEnabled) {
+      this.isInvalidated = true;
+    }
+    switch (cmd.type) {
+      case "UNDO":
+      case "REDO":
+      case "ACTIVATE_SHEET":
+        if (this.isEnabled) {
+          this.isInvalidated = true;
+        }
+        break;
+    }
+  }
+
+  finalize() {
+    if (this.isInvalidated) {
+      this.isInvalidated = false;
+      this.computeFingerprints();
+    }
+  }
+
+  enable() {
+    this.isEnabled = true;
+    this.computeFingerprints();
+  }
+
+  disable() {
+    this.isEnabled = false;
+    this.map = new PositionMap();
+  }
+
+  private computeFingerprints() {
+    this.map = new PositionMap();
+    const fingerprints = new PositionMap<Fingerprint>();
+    const allFingerprints = new Set<Fingerprint>();
+    const activeSheetId = this.getters.getActiveSheetId();
+    const cells = this.getters.getCells(activeSheetId);
+    for (const cellId in cells) {
+      const fingerprint = this.computeFingerprint(cells[cellId]);
+      if (!fingerprint) {
+        continue;
+      }
+      allFingerprints.add(fingerprint);
+      const position = this.getters.getCellPosition(cellId);
+      fingerprints.set(position, fingerprint);
+    }
+    const fingerprintColors = this.assignColors(allFingerprints);
+    for (const [position, fingerprint] of fingerprints.entries()) {
+      const color = fingerprintColors[fingerprint];
+      this.map.set(position, color);
+      this.colorSpreadZone(position, color);
+    }
+  }
+
+  private colorSpreadZone(position: CellPosition, fingerprintColor: Color) {
+    const spreadZone = this.getters.getSpreadZone(position);
+    if (!spreadZone) {
+      return;
+    }
+    const sheetId = position.sheetId;
+    for (let row = spreadZone.top; row <= spreadZone.bottom; row++) {
+      for (let col = spreadZone.left; col <= spreadZone.right; col++) {
+        const spreadPosition = { sheetId, col, row };
+        this.map.set(spreadPosition, fingerprintColor);
+      }
+    }
+  }
+
+  private assignColors(fingerprints: Set<Fingerprint>): Record<Fingerprint, Color> {
+    const fingerprintColors = {
+      [NULL_VECTOR_FINGERPRINT]: "#909090",
+      [NUMBER_FINGERPRINT]: "#D9D9D9",
+      [BOOLEAN_FINGERPRINT]: "#E9E9E9",
```

Comment:
Because numbers (and arguably booleans) can be considered as data while strings can be much more, like titles etc.

Coloring strings would probably add more noise than anything else.